pkg/strategy/production.go

```diff
@@ -22,98 +20,96 @@ import (
-	queryDep := inject.OAuthProxy(c.jaeger, query.Get())
+	queryDep := inject.OAuthProxy(jaeger, query.Get())
 
 	// add the deployments
-	os = append(os, cDep, queryDep)
+	c.deployments = []appsv1.Deployment{*collector.Get(), *inject.OAuthProxy(jaeger, query.Get())}
 
-	if ds := agent.Get(); nil != ds {
-		os = append(os, ds)
+	// add the daemonsets
+	if ds := agent.Get(); ds != nil {
+		c.daemonSets = []appsv1.DaemonSet{*ds}
 	}
 
 	// add the services
 	for _, svc := range collector.Services() {
-		os = append(os, svc)
+		c.services = append(c.services, *svc)
 	}
 
 	for _, svc := range query.Services() {
-		os = append(os, svc)
+		c.services = append(c.services, *svc)
 	}
 
 	// add the routes/ingresses
 	if viper.GetString("platform") == v1alpha1.FlagPlatformOpenShift {
-		if q := route.NewQueryRoute(c.jaeger).Get(); nil != q {
-			os = append(os, q)
+		if q := route.NewQueryRoute(jaeger).Get(); nil != q {
+			c.routes = append(c.routes, *q)
 		}
 	} else {
-		if q := ingress.NewQueryIngress(c.jaeger).Get(); nil != q {
-			os = append(os, q)
+		if q := ingress.NewQueryIngress(jaeger).Get(); nil != q {
+			c.ingresses = append(c.ingresses, *q)
 		}
 	}
 
-	if isBoolTrue(c.jaeger.Spec.Storage.SparkDependencies.Enabled) {
-		if cronjob.SupportedStorage(c.jaeger.Spec.Storage.Type) {
-			os = append(os, cronjob.CreateSparkDependencies(c.jaeger))
+	if isBoolTrue(jaeger.Spec.Storage.SparkDependencies.Enabled) {
+		if cronjob.SupportedStorage(jaeger.Spec.Storage.Type) {
+			c.cronJobs = append(c.cronJobs, *cronjob.CreateSparkDependencies(jaeger))
 		} else {
-			logrus.WithField("type", c.jaeger.Spec.Storage.Type).Warn("Skipping spark dependencies job due to unsupported storage.")
+			logrus.WithField("type", jaeger.Spec.Storage.Type).Warn("Skipping spark dependencies job due to unsupported storage.")
 		}
 	}
 
 	var indexCleaner *batchv1beta1.CronJob
-	if isBoolTrue(c.jaeger.Spec.Storage.EsIndexCleaner.Enabled) {
-		if strings.EqualFold(c.jaeger.Spec.Storage.Type, "elasticsearch") {
-			indexCleaner = cronjob.CreateEsIndexCleaner(c.jaeger)
-			os = append(os, indexCleaner)
+	if isBoolTrue(jaeger.Spec.Storage.EsIndexCleaner.Enabled) {
+		if strings.EqualFold(jaeger.Spec.Storage.Type, "elasticsearch") {
+			indexCleaner = cronjob.CreateEsIndexCleaner(jaeger)
+			c.cronJobs = append(c.cronJobs, *indexCleaner)
 		} else {
-			logrus.WithField("type", c.jaeger.Spec.Storage.Type).Warn("Skipping Elasticsearch index cleaner job due to unsupported storage.")
+			logrus.WithField("type", jaeger.Spec.Storage.Type).Warn("Skipping Elasticsearch index cleaner job due to unsupported storage.")
 		}
 	}
 
-	if storage.ShouldDeployElasticsearch(c.jaeger.Spec.Storage) {
+	if storage.ShouldDeployElasticsearch(jaeger.Spec.Storage) {
 		es := &storage.ElasticsearchDeployment{
-			Jaeger: c.jaeger,
+			Jaeger: jaeger,
 		}
-		objs, err := es.CreateElasticsearchObjects(cDep.Spec.Template.Spec.ServiceAccountName, queryDep.Spec.Template.Spec.ServiceAccountName)
```

Comment:
@jpkrohling this PR broke Elasticsearch functionality, note that `es.CreateElasticsearchObjects` was returning ES CR in the object array. Now the ES CR is not being processed.